src/peppr/match.py

```diff
@@ -38,158 +43,148 @@ class GraphMatchWarning(UserWarning):
     pass
 
 
-def find_matching_atoms(
+class UnmappableEntityError(Exception):
+    """
+    This exceptions is raised, if the reference and pose structure contain
+    entities that cannot be mapped to each other.
+    """
+
+    pass
+
+
+def find_optimal_match(
     reference: struc.AtomArray,
     pose: struc.AtomArray,
     min_sequence_identity: float = 0.95,
+    use_heuristic: bool = True,
+    max_matches: int | None = None,
 ) -> tuple[NDArray[np.int_], NDArray[np.int_]]:
     """
-    Find the optimal atom order for each pose that minimizes the RMSD to the reference.
+    Find the atom indices for the given reference and pose structure that brings these
+    structure into a corresponding order that minimizes the RMSD between them.
 
     Parameters
     ----------
-    reference : struc.AtomArray, shape=(n,)
+    reference : AtomArray, shape=(p,)
         The reference structure.
-    pose : struc.AtomArray, shape=(n,)
+    pose : AtomArray, shape=(q,)
         The pose structure.
     min_sequence_identity : float
         The minimum sequence identity between two chains to be considered the same
         entity.
+    use_heuristic : bool or int
+        Whether to employ a fast heuristic [1]_ to find the optimal chain permutation.
+        This heuristic represents each chain by its centroid, i.e. instead of
+        exhaustively superimposing all atoms for each permutation, only the centroids
+        are superimposed and the closest match between the reference and pose is
+        selected.
+    max_matches : int, optional
+        The maximum number of atom mappings to try, if the `use_heuristic` is set to
+        ``False``.
 
     Returns
     -------
-    reference_order, : np.array, shape=(n,), dtype=int
-        The atom order that should be applied to `reference`.
-    pose_order : np.array, shape=(n,), dtype=int
-        The atom order that should be applied to `pose`.
+    reference_indices : np.array, shape=(n,), dtype=int
+        The atom indices that should be applied to `reference`.
+    pose_indices : np.array, shape=(n,), dtype=int
+        The atom indices that should be applied to `pose`.
```

Comment:
Is the point of returning both here so that it's up to the caller to determine which structure gets re-ordered? Otherwise one might naively expect that we only one need a single array of indices.

Reply:
This function returns the common subset of atoms from both input structures. This means, if both structures miss some atoms, both returned index arrays will contain gaps within them

Reply:
Understood, and this makes a lot of sense!